source/scripts/RecipePage.js

```diff
@@ -0,0 +1,230 @@
+        text-align: center;
+        font-weight: bold;
+        margin-top: 80px;
+    }
+  
+    #clear-checkboxes{
+      position: relative;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+      margin-top: 40px;
+    }
+  
+    /*-- custom checkbox style --*/
+    
+    input[class="ingredients-custom-checkbox"] + label,
+    input[class="ingredients-custom-checkbox"] + label::before
+    {
+        display: inline-block;
+        vertical-align: middle;
+        
+    }
+  
+    input[class="ingredients-custom-checkbox"]
+    {
+        opacity: 0; /*hide the actual checkbox*/
+        position: absolute;
+        transform: scale(2);
+    }
+  
+    input[class="ingredients-custom-checkbox"]:hover
+    {
+      cursor: pointer;
+    }
+      
+    input[class="ingredients-custom-checkbox"] + label::before
+    {
+      /*custom new checkbox*/
+        content: "";
+        width: 15px;
+        height: 15px;
+        margin: 0 4px 0 0;
+        line-height: 14px;
+        text-align: center;
+        border: solid 3px var(--primary);
+    }
+    
+      input[class="ingredients-custom-checkbox"]:checked + label::before
+    {
+        content: "\\2714"; /* tick sign */
+        font-size: 15px;
+    }
+  
+    #ingredients-list > ul > ol > label{
+      color: black;
+    }
+    /*-----------------------------------*/
+    
+    #instructions > ol > li{
+      color: black;
+    }
+  
+    #ingredients-list{
+        /* text-align: center; */
+        font-style: normal;
+        font-weight: normal;
+  
+    }
+  
+    #ingredients-list > button{
+      border: 0;
+      background-color: inherit;
+      font: inherit;
+      text-align: center;
+      padding: 20px;
+    }
+  
+    #ingredients-list> button:hover{
+      cursor: pointer;
+      background: #eee;
+    }
+      `;
+
+
+    this.shadowRoot.append(style, container);
+  }
+
+  set data(data) {
+    if (data == null) {
+      console.log("Error: no data exists");
+      return;
+    }
+
+    this.shadowRoot.querySelector("article").innerHTML = `
+      <section class="middle">
+        <header class="header">
+          <h1></h1>
+            <img id="bookmark-icon" onclick="toggleBookMark(this)" src="./img/icons/bookmark-empty.svg"" width="56" height="56">
+         </header>
+          <h1></h1>
+          <div id="ingredients-list">
+              <h3>INGREDIENTS</h3>
+              <ul style="list-style-type: none;">
+              </ul>
+              <button id="clear-checkboxes" onclick="clearCheckBoxes()">CLEAR CHECKBOXES</button>
+          </div>
+          <div id="instructions">
+              <h3>INSTRUCTIONS</h3>   
+              <ol>
+              </ol>
+          </div>
+      </section>
+      `;
+
+    // Header - title
+
+    this.shadowRoot.querySelector(".header > h1").innerHTML = data["title"];
+
+    //get ingredient list
+    const ingredients = getIngredients(data);
+    ingredients.forEach(ingredient => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8414136)
```suggestion
    ingredients.forEach((ingredient) => {
```